src/index.ts

```diff
@@ -1404,15 +1421,30 @@ export class ValidationService {
             }
         } else if (mutation.type === 'attributes') {
             if (mutation.target instanceof HTMLElement) {
-                const oldValue = mutation.oldValue ?? '';
-                const newValue = mutation.target.attributes[mutation.attributeName]?.value ?? '';
-                this.logger.log("Attribute '%s' changed from '%s' to '%s'",
-                    mutation.attributeName,
-                    oldValue,
-                    newValue,
-                    mutation.target);
-                if (oldValue !== newValue) {
-                    this.scan(mutation.target);
+                const attributeName = mutation.attributeName;
+
+                // Special case for disabled.
+                if (attributeName === 'disabled') {
+                    const target = mutation.target as ValidatableElement;
+                    if (target.disabled) {
+                        this.remove(target);
+                        this.resetField(this.getElementUID(target));
```

Comment:
I don't think we want to `remove()`, but I do like the observer resetting validation messages if we notice it becomes disabled. 👍

Reply:
I believe the remove is necessary otherwise we highlight the field in green when it is valid.

<img width="548" alt="image" src="https://github.com/haacked/aspnet-client-validation/assets/19977/948ee2ae-b63d-4eb1-b75f-f9312dbc72e1">

Let me look into that though. Maybe I just need to prevent that code from running when the field is disabled too.